simulator/syncer/syncer.go

```diff
@@ -0,0 +1,339 @@
+package syncer
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+// GVRs is a list of GroupVersionResource that we want to sync.
+// Note that this order matters - When first importing resources, we want to sync namespaces first, then priorityclasses, storageclasses...
+var GVRs = []schema.GroupVersionResource{
```

Comment:
This variable is public, so users can modify it.
In what cases would users need to change this variable?

Reply:
I think users would have a scheduler that schedules pods based on resources other than default ones.

> All resources involved in the scheduling should be synced.
> And, we should make it configurable to select which resources to sync, given everyone could have a different scheduler plugin which schedules Pods based on anything.

https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/327

Reply:
Understood. Thank you for the explanation!